pkg/controller/deployment/util/deployment_util.go

```diff
@@ -132,6 +132,18 @@ func GetDeploymentCondition(status extensions.DeploymentStatus, condType extensi
 	return nil
 }
 
+// TODO: remove the duplicate
+// GetDeploymentConditionInternal returns the condition with the provided type.
+func GetDeploymentConditionInternal(status internalextensions.DeploymentStatus, condType internalextensions.DeploymentConditionType) *internalextensions.DeploymentCondition {
```

Comment:
> I thought the intention with respect to utilities used by clients was that we are going to have utilities use internal types and clients should run conversions if they wanted to use our utilities, is that correct?

It's the other way around. Because and all controllers are refactored to use versioned API, and they are performance-sensitive, so I converted the utilities to versioned. Kubectl still uses internal version, so kubectl needs to convert deployment to versioned before passing it to the utilities.

Regarding this `GetDeploymentConditionInternal`. Given that
* its caller pkg/kubectl/rollout_status.go doesn't call any other deployment utility function
* this function is small

I think it's more maintainable and readable to just duplicate the function, instead of calling conversion in rollout_status.go.